src/Features/OnboardingSetUpShipping.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Onboarding - set up shipping.
+ *
+ * @package Woocommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use \Automattic\WooCommerce\Admin\PluginsHelper;
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Notes_Review_Shipping_Settings;
+
+/**
+ * This contains logic for setting up shipping when the profiler completes.
+ */
+class OnboardingSetUpShipping {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action(
+			'woocommerce_onboarding_profile_completed',
+			array(
+				__CLASS__,
+				'on_onboarding_profile_completed',
+			)
+		);
+	}
+
+	/**
+	 * Set up shipping.
+	 */
+	public static function on_onboarding_profile_completed() {
+		if ( ! self::is_physical_selected_as_product_type() ) {
+			return;
+		}
+
+		if ( self::are_there_existing_shipping_zones() ) {
+			return;
+		}
+
+		self::set_up_free_local_shipping();
+		WC_Admin_Notes_Review_Shipping_Settings::possibly_add_note();
+		wc_admin_record_tracks_event( 'shipping_automatically_set_up' );
+	}
+
+	/**
+	 * Are there existing shipping zones?
+	 *
+	 * @return bool
+	 */
+	private static function are_there_existing_shipping_zones() {
+		$zone_count = count( \WC_Shipping_Zones::get_zones() );
+
+		return $zone_count > 0;
+	}
+
+	/**
+	 * Is 'physical' selected as a product type?
+	 *
+	 * @return bool
+	 */
+	private static function is_physical_selected_as_product_type() {
```

Comment:
Nitpick: can we rename this to `has_physical_products`?

Reply:
done